xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterLocationStateComputer.java

```diff
@@ -44,33 +48,53 @@ public class ScopeNotificationFilterLocationStateComputer
     @Inject
     private ScopeNotificationFilterPreferencesGetter preferencesGetter;
 
-    private enum WatchedState
-    {
-        WATCHED,
-        NOT_WATCHED,
-        UNKNOWN
-    }
-
     /**
-     * @param filterPreferences the collection of all preferences to take into account
-     * @param location a location
-     * @return {@code true} if the location is watched by the user, for any kind of event types or format.
+     * Compute the watch state for the given location, based on the given preferences, for all formats and all event
+     * types.
+     *
+     * @param filterPreferences the list of filter preferences to consider for computing the watch state
+     * @param location the location for which to compute the watch state
+     * @return the watch state for all formats and all event types
      */
-    public boolean isLocationWatched(Collection<NotificationFilterPreference> filterPreferences,
-        EntityReference location)
+    // TODO: This should be probably cached info.
```

Comment:
Is this manipulating only the filter preference table or also stuff like getting the user document ?

Reply:
Actually neither in this code: it's only a computation based on the given collection of `NotificationFilterPreference` so it doesn't access any storage.

Reply:
OK, cool.